components/concept/fetch/src/main/java/mozilla/components/concept/fetch/Request.kt

```diff
@@ -67,6 +67,11 @@ data class Request(
              */
             fun fromFile(file: File): Body = Body(file.inputStream())
 
+            /**
+             * Create a [Body] from the provided [ByteArray].
+             */
+            fun fromByteArray(array: ByteArray) = Body(array.inputStream())
```

Comment:
This change may affect a-s which compiles against concept-fetch when it lands.

Reply:
Even with this method being fully implemented here and not being used?

Reply:
It may be possible. When the `Request` constructor had a new argument with a default value added to it, the class signature changed there. I'm honestly not sure how the bytecode signature's may affect it here when it's only a static function. 🤔 

Maybe I'm being too cautious? 😅

Reply:
Let's work on breaking up this dependency cycle (not like now, but soon)! I guess there's no reason why AS can't have their own, super minimal, interface for requests and we map that to `concept-fetch` here?

Reply:
concept-concept-fetch! 

Yeah that might be the next thing to investigate.

Reply:
Glean pretty much solved this by having the `PingUploader` interface, that has multiple implementations and we have a `concept-fetch` mapping.

* https://github.com/mozilla/glean/blob/main/docs/user/general-api.md
* https://github.com/mozilla/glean/blob/main/glean-core/android/src/main/java/mozilla/telemetry/glean/net/PingUploader.kt#L65-L77
* https://github.com/mozilla-mobile/android-components/blob/master/components/service/glean/src/main/java/mozilla/components/service/glean/net/ConceptFetchHttpUploader.kt#L29

Reply:
Thank you, I didn't know about this! I'll file an issue then to look into this.

Reply:
> Let's work on breaking up this dependency cycle (not like now, but soon)! I guess there's no reason
> why AS can't have their own, super minimal, interface for requests and we map that to concept-fetch here?

FWIW, I agree with the suggestion here - break the cycle by having appservices define an interface that it uses, and having the a-c RustHttp integration map concept-fetch into that interface.